pkg/diagnostics/systemd/unit_status.go

```diff
@@ -0,0 +1,128 @@
+package systemd
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"runtime"
+
+	"github.com/openshift/origin/pkg/diagnostics/log"
+	"github.com/openshift/origin/pkg/diagnostics/types"
+)
+
+// UnitStatus is a Diagnostic to check status of systemd units that are related to each other.
+type UnitStatus struct {
+	SystemdUnits map[string]types.SystemdUnit
+}
+
+const UnitStatusName = "UnitStatus"
+
+func (d UnitStatus) Name() string {
+	return UnitStatusName
+}
+
+func (d UnitStatus) Description() string {
+	return "Check status for related systemd units"
+}
+func (d UnitStatus) CanRun() (bool, error) {
+	if runtime.GOOS == "linux" {
+		if _, err := exec.LookPath("systemctl"); err == nil {
+			return true, nil
+		}
+	}
+
+	return false, errors.New("systemd is not present on this host")
+}
+func (d UnitStatus) Check() types.DiagnosticResult {
+	r := types.NewDiagnosticResult(UnitStatusName)
+
+	unitRequiresUnit(r, d.SystemdUnits["openshift-node"], d.SystemdUnits["iptables"], nodeRequiresIPTables)
+	unitRequiresUnit(r, d.SystemdUnits["openshift-node"], d.SystemdUnits["docker"], `Nodes use Docker to run containers.`)
+	unitRequiresUnit(r, d.SystemdUnits["openshift-node"], d.SystemdUnits["openvswitch"], sdUnitSDNreqOVS)
+	unitRequiresUnit(r, d.SystemdUnits["openshift-master"], d.SystemdUnits["openvswitch"], `Masters use openvswitch for access to cluster SDN networking`)
+	// all-in-one networking *could* be simpler, so fewer checks
+	unitRequiresUnit(r, d.SystemdUnits["openshift"], d.SystemdUnits["docker"], `Nodes use Docker to run containers.`)
```

Comment:
So far they've changed every month or two.

Also have to update for AEP...

Eventually diagnostics could probably be part of an enterprise e2e test?